zebrad/src/components/tracing/endpoint.rs

```diff
@@ -52,42 +63,53 @@ impl TracingEndpoint {
     #[cfg(feature = "filter-reload")]
     #[allow(clippy::unwrap_in_result)]
     pub fn init_tokio(&mut self, tokio_component: &TokioComponent) -> Result<(), FrameworkError> {
-        use hyper::{
-            service::{make_service_fn, service_fn},
-            Server,
-        };
-
         let addr = if let Some(addr) = self.addr {
             addr
         } else {
             return Ok(());
         };
 
-        let service =
-            make_service_fn(|_| async { Ok::<_, hyper::Error>(service_fn(request_handler)) });
-
         info!("Trying to open tracing endpoint at {}...", addr);
+
+        let svc =
+            tower::service_fn(|req: Request<Incoming>| async move { request_handler(req).await });
```

Comment:
Optional/Nitpick: We can build a hyper service instead since we're using a function anyway

```suggestion
        let svc = hyper::service::service_fn(|req: Request<Incoming>| async move {
            request_handler(req).await
        });
```

Reply:
That's good, thanks. done in https://github.com/ZcashFoundation/zebra/pull/8692/commits/62a8d7191462d6b8393e518e4d646a09542f3699